h/atom_feed.py

```diff
@@ -0,0 +1,159 @@
+import re
+import urlparse
+import cgi
+import datetime
+
+import pyramid.i18n
+
+_ = pyramid.i18n.TranslationStringFactory(__package__)
+
+
+def _username_from_annotation(annotation):
+    """Return the username from the given annotation.
+
+    For example if the annotation contains
+    {"user": "acct:seanh@hypothes.is", ...} then return "seanh".
+
+    """
+    match = re.match(r'^acct:([^@]+)@(.*)$', annotation["user"])
+    username, domain = match.groups()
+    return username
+
+
+def _created_day_string_from_annotation(annotation):
+    """Return a simple created day string for an annotation.
+
+    Returns a day string like '2015-03-11' from a Hypothesis API "created"
+    datetime string like '2015-03-11T10:43:54.537626+00:00'.
+
+    """
+    return datetime.datetime.strptime(
+        annotation["created"][:10], "%Y-%m-%d").strftime("%Y-%m-%d")
```

Comment:
A clear violation of [RFC2550](https://tools.ietf.org/html/rfc2550).

P.S. Ignore me.